models/gpg_key.go

```diff
@@ -5,40 +5,77 @@
 package models
 
 import (
+	"bytes"
 	"fmt"
 	"strings"
 	"time"
 
 	"golang.org/x/crypto/openpgp"
+	"golang.org/x/crypto/openpgp/packet"
 
 	"code.gitea.io/gitea/modules/log"
 	"github.com/go-xorm/xorm"
 )
 
+/*
+// GPGKeyEmail represents a GPG email in a GPGKey.
+type GPGKeyEmail struct {
+	Email    string `json:"email"`
+	Verified bool   `json:"verified"`
+}
+*/
+
 // GPGKey represents a GPG key.
 type GPGKey struct {
-	ID      int64  `xorm:"pk autoincr"`
-	OwnerID int64  `xorm:"INDEX NOT NULL"`
-	KeyID   string `xorm:"INDEX TEXT NOT NULL"`
-	//PrimaryKeyID string    `xorm:"TEXT"`
-	Content     string    `xorm:"TEXT NOT NULL"`
-	Created     time.Time `xorm:"-"`
-	CreatedUnix int64
-	Added       time.Time `xorm:"-"`
-	AddedUnix   int64
+	ID                int64     `xorm:"pk autoincr"`
+	OwnerID           int64     `xorm:"INDEX NOT NULL"`
+	KeyID             string    `xorm:"INDEX TEXT NOT NULL"`
+	PrimaryKeyID      string    `xorm:"TEXT"`
+	Content           string    `xorm:"TEXT NOT NULL"`
+	Created           time.Time `xorm:"-"`
+	CreatedUnix       int64
+	Expired           time.Time `xorm:"-"`
+	ExpiredUnix       int64
+	Added             time.Time `xorm:"-"`
+	AddedUnix         int64
+	SubsKey           []*GPGKey `xorm:"-"`
+	Emails            []string
```

Comment:
After searching and testing, I found that github only accept key with a allready verified email. So []models.EmailAddress is definitly perfect for that.